pkgs/tools/games/scarab/scaling-settings.bash

```diff
@@ -0,0 +1,17 @@
+# Keep existing value if it is already non-empty
+if [[ -z "${AVALONIA_GLOBAL_SCALE_FACTOR-}" ]]; then
+    # Attempt to get GNOME desktop interface scaling factor
+    AVALONIA_GLOBAL_SCALE_FACTOR="$(gsettings get org.gnome.desktop.interface scaling-factor | cut --delimiter=' ' --fields=2)"
+fi
+
+if [[ -z "${AVALONIA_GLOBAL_SCALE_FACTOR-}" ]]; then
+    # Attempt to get scaling factor from X FreeType DPI setting
+    dpi="$(xrdb -get Xft.dpi)"
+    if [[ -n "$dpi" ]]; then
+        AVALONIA_GLOBAL_SCALE_FACTOR=$(echo "scale=2; $dpi/96" | bc)
```

Comment:
Should we use bash arithmetic here to avoid the `bc` dependency?

Reply:
Bash only does integer arithmetic, so it would break in case of non-integer scaling. We should probably support non-integer scaling if possible.

Reply:
Ah sorry, I'm so used to doing it in zsh that I forgot it wasn't standard.